### PR TITLE
fix: fix children with same key, uniqueId

### LINKED ISSUE #757: React "two children with same key" warning from `<Trans>` macro
**Describe the bug**
When using the `<Trans>` macro, if the language string contains two instances of the same variable, _and_ that variable's value is a JSX element, Lingui generates two sibling elements with the same `key` prop. This causes React to complain, because the `key` prop is required to be unique for sibling elements.

**To Reproduce**
Here is a Storybook story that reproduces the issue:

```
import React from 'react';
import { Trans } from '@lingui/macro';
import { storiesOf } from '@storybook/react';
import { I18nProvider } from '@lingui/react';

const foo = <span>bar</span>;

storiesOf('lingui', module).add('two children with the same key', () => (
  <I18nProvider language="en">
    <Trans>
      {foo} {foo}
    </Trans>
  </I18nProvider>
));
```

**Expected behavior**
The story displays "bar bar" when viewed in the browser, with no warnings or errors in the JS console.

**Actual behavior**
The story displays correctly in the browser, but it generates React elements with the same key of `0`, and React prints a warning about that to the JS console:

```
Warning: Encountered two children with the same key, `0`. Keys
should be unique so that components maintain their identity across
updates. Non-unique keys may cause children to be duplicated
and/or omitted — the behavior is unsupported and could change in a
future version.
    in Render (created by Trans)
    in Trans (created by I18n)
    in I18n (created by WithI18n)
    in WithI18n (at lingui.stories.tsx:10)
    in I18nProvider (at lingui.stories.tsx:9)
    in storyFn
    in ErrorBoundary
```
When the rendered components are viewed using the React Dev Tools browser plugin, I can see this hierarchy of components:

* `WithI18n`
    * `I18n`
        * `Trans`
            * `Render`
                * `span key="0"`
                * `span key="0"`

**Additional context**
- jsLingui version 2.9.2
- Babel version 7.9.0
- Create React App (react-scripts 3.4.3)

### PR DESCRIPTION
**Pull request by semoal**
Fixes #757 

It's using a simpe lodash uniqueId function, just a const to keep the past id's 👍🏻 
I've introduced a little test to avoid a regression to this

### CODE REVIEW

File: packages/react/src/format.ts [hunk 1]
```diff
@@ -105,4 +106,14 @@ function getElements(parts) {
...
+function uniqueId(prefix: "$lingui$" = "$lingui$") {
+  if (!idCounter[prefix]) {
+    idCounter[prefix] = 0
+  }
+
+  const id =++idCounter[prefix]
+
+  return `${prefix}_${id}`
+}
+
```
Comment (tricoder42) [inline packages/react/src/format.ts:118]: I would prefer simple closure before global variable:

```suggestion
const makeCounter = (count = 0, prefix = "") => () => `${prefix}${count++}`
```

Then we can create a counter inside `formatFunction`:

```js
const uniqueId = makeCounter(0, '$lingui$')
```

I think it should be safe to create the counter on every call of `formatElements`. The problem was that elements without an index (variables) were getting the same id.

**Review by tricoder42 — COMMENTED** [review]
Good idea, just one suggestion 👇

File: packages/react/src/format.ts (hunk 1 above)
Comment (semoal) [reply packages/react/src/format.ts]: Yes, absolutly, i've changed to match your suggestion!!

**Comment by semoal** [comment]
Changes made @tricoder42, we can :shipit:

**Comment by tricoder42** [comment]
@semoal Pretty good! Thanks 🚀

**Pull request merged**